src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -125,6 +125,14 @@ namespace winrt::TerminalApp::implementation
             p.SetActionMap(_settings.ActionMap());
         }
 
+        if (_extensionPalette)
+        {
+            // the extension palette had been loaded with the previous settings
+            // reload it with the new settings
+            _extensionPalette = nullptr;
```

Comment:
danger: because I still haven't fixed #11522, this will blow away the entire palette every time the user changes the keyboard layout